### ISSUE TITLE
Sometimes my image doesn't load

### ISSUE THREAD

**Issue by shamoons** [TeamWertarbyte/material-ui-image#51]
```
<Image src={tile.src} imageStyle={{ height: tile.height, width: tile.width }} color='transparent' />
```

It stays spinning forever:
<img width="227" alt="image" src="https://user-images.githubusercontent.com/1019677/83330642-8332ee00-a25e-11ea-85b2-21b324d22a29.png">

even though the image was requested in the `Network tab` of my inspector:
<img width="227" alt="image" src="https://user-images.githubusercontent.com/1019677/83330649-8d54ec80-a25e-11ea-8e42-40093152816e.png">

**Comment by developerjoe** [comment]
I'm wondering if this is the same issue as #43. I encounter this problem with using NextJS. If I wrap the Image component in the NoSsr component the image is successfully loaded.

**Comment by ydax** [comment]
Yep, this is an issue for us, too.

**Comment by dvargas92495** [comment]
I think the `img`'s `onLoad` function isn't being called, since we are loading the image during server side rendering instead of after the javascript is loaded. So basically, this line is not firing: https://github.com/TeamWertarbyte/material-ui-image/blob/43c67fcc3b8eb1f5c91bcc04906b3e88e9019fe1/src/components/Image/Image.js#L88 

This comment seems related: https://github.com/hzdg/react-imageloader/issues/16#issuecomment-403163664

I think I was able to get around this with the following approach:
```jsx
const url = "/static/image.png"

const Index = () => {
  const imageRef = useRef<typeof Image>(null);
  useEffect(() => {
    // @ts-ignore
    if (!imageRef.current.state.imageLoaded) {
      // @ts-ignore
      imageRef.current.handleLoadImage();
    }
  }, [imageRef]);
  return (
    <div>
        <Image
          src={data.landingImage.publicURL}
          aspectRatio={WIDTH / HEIGHT}
          // @ts-ignore
          ref={imageRef}
        />
    </div>
  );
```

Basically, using a ref to call handle load image manually. This is not ideal because 1) I'm littering some ts-ignore's since the type/ref is not supported and 2) this will be unexpected behavior if the image does fail loading for any reason. But it's a good workaround for now.

**Comment by dvargas92495** [comment]
> Can you give an example?

I just tested using material-ui's component and it seemed to work also, which is much better than the workaround I previously posted: https://material-ui.com/components/no-ssr/

**Comment by ydax** [comment]
@dvargas92495 this solution's working well for us. Thanks!

**Comment by jbgarr** [comment]
I was having a similar issue, but it was only happening in IE11. In our situation we have a gallery component that iterates over an array of image URLs and renders an `Image` component for each, passing in the URL as the `src` prop.

Long story short, we were using index as the key when mapping over the image URLs (I know, bad practice!), but once I switched the index out for a unique identifier it worked as expected. After that change, when the image URLs changed (to view a different set of images), all of the `Image` components re-rendered and loaded the new images as expected.

It sounds like there is another possible work around using the MUI `NoSsr` component, but I wanted to throw this out there as well in case it's helpful for anyone.

**Comment by maciejkopecpl** [comment]
I just faced the same issue. @dvargas92495 seems to be working for me.

**Comment by leMaik** [comment]
Sorry this took so long. :sweat_smile: I implemented [this solution](https://stackoverflow.com/a/54376873), so as of v3.2.4 everything should just work. :+1: Please create a new issue if it doesn't.
